### ISSUE TITLE
Adding local node

### ISSUE THREAD

**Issue by alex3s** [scala-network/ScalaVault#1]
Hi, 
Scala is such a exciting software!
I want to use my local node.
When adding the local node to the list on android 8  192.168.1.1:11812 Test: works well, ping below 100ms, after pressing OK the node appears on the list, but after going back to the wallet(s) list, the node is not used, and disappears from the nodes list. 
The same happens on Android10.
Cheers!

**Comment by Mas7erMind** [comment]
Ok we are checking this. The way the node management works is that the app chooses the best available node, so your node may not be used automatically. We could implement a way to force the use of a node, especially the custom ones. But there seems to be a bug if your node is not displayed anymore in the list of node, we'll check that.

**Comment by alex3s** [comment]
Actually, the sorting of the node list, and removing the dead nodes is another issue which can be implemented better this way.
What I mean is this: yesterday I connected to a local network with with no outside connection to the internet. My internet modem which was sharing it's connection through a local hotspot was down for a moment. Suddenly all the nodes disappeared from the list, I guess because none of them was reachable. After connecting to the Internet they did not reappear, so I could not open the wallet.  Only removing the SVault from the apps on the screen or restarting the phone resetted the list. What would be a better solution? Unreachable nodes appear greyed out on the bottom of the list, pinged alive when and active again during opening of the wallet.What is your opinion?

**Comment by Mas7erMind** [comment]
Have you tired refreshing the list of nodes when your connection came back? You can do so by swiping from top to down in the node list view.

**Comment by alex3s** [comment]
Hi, and thank you for your answer. I can confirm that swiping down on Android 8 refreshes the status of the nodes: i.e. those with icons crossed out "CONNECTION ERROR" are updated to "Last block updated: ..." and normal icon. Unfortunately those which disappeared completely from the list during initial startup of the app (or maybe when running in the background), they do not appear again when swiping down, gone from the list until the app is wiped out from memory and restarted. I think it's a minor nuisance which I did not anticipate, not something crucial. Thanks again for updating the source to include the adding of a node functionality! Cheers!

**Comment by Mas7erMind** [comment]
Ok thanks for the feedback! Actually, the nodes that are not reachable will appear at startup but disappear from the list when you swipe down (refresh). It's a questionable behavior, but as you said it may be more a nuisance than a bug. This comes from monerujo and we kept it this way. Anyway, I fixed the bug you mentioned, it will be available in the next release. I really appreciate your comments, thanks for using our app! 👍

**Comment by alex3s** [comment]
@Mas7erMind, Maybe it would be a good idea to change the message from 
`<string name="prompt_daemon_missing">**Node must be set!**</string>`
in strings.xml
To more user friendly: "**Connect to the Internet or restart the app to refresh available remote nodes!**"
or something like this ?

**Comment by Mas7erMind** [comment]
@alex3s Thanks for all your comments, it helps make our app better!

I fixed many things in the node management routine:
- Fixed userdefined nodes not being saved
- Allow the user to manually select the remote node he wants to use
- Don't hide unreachable nodes
- Add a hint message (swipe down to refresh)
- Replace error message

**Comment by alex3s** [comment]
@Mas7erMind, @ahmyi, @hayzamjs, @donovansolms I have created a lightweight scala pool to serve mobile phone mining which is a very exciting (and unique imho) feature of scala. It's located here: https://github.com/alex3s/scala-puddle

I was able to compile it on a Linux machine as well as on Android inside Termux, accompanied by Scala node on the same phone. 

It runs, however all the shares are invalid (wrong hashes reported). Maybe it's Panthera used interchangeably with RandomX. So far I couldn't figure it out. Any ideas what can be wrong? I didn't know where to post my question for public support :)
